conversion/indico_conversion/plugin.py

```diff
@@ -115,10 +128,13 @@ def _form_validated(self, form, **kwargs):
         g.convert_attachments_pdf = form.ext__convert_to_pdf.data
 
     def _attachment_created(self, attachment, **kwargs):
-        if not g.get('convert_attachments_pdf') or attachment.type != AttachmentType.file:
+        if not g.get('convert_attachments_pdf'):
             return
-        ext = os.path.splitext(attachment.file.filename)[1].lstrip('.').lower()
-        if ext not in self.settings.get('valid_extensions'):
+        if attachment.type == AttachmentType.file:
+            ext = os.path.splitext(attachment.file.filename)[1].lstrip('.').lower()
+            if ext not in self.settings.get('valid_extensions'):
+                return
+        elif not attachment.link_url.startswith('https://docs.google.com/'):
```

Comment:
Would it make sense to match the exact pattern of the supported links? Right now you attempt to convert any link starting with that prefix, but if it's some garbage it'd fail multiple times until the retries are exhausted, instead of not being queued for conversion at all.

Reply:
Does this gain us much? This would catch the case where we have `https://docs.google.com/<garbage>/<valid fileid>` but it cannot catch problems with `<garbage fileid>` (which I think is a more likely error). Plus this way, we are less vulnerable if google add more doc types (which can be converted) and/or change the path

Reply:
Okay, I think I have improved this a little.